packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js

```diff
@@ -86,14 +82,40 @@ class UniswapPriceFeed extends PriceFeedInterface {
 
   // Not part of the price feed interface. Can be used to pull the uniswap price at the most recent block.
   getLastBlockPrice() {
-    return this.lastBlockPrice && this.convertPoolDecimalsToPriceFeedDecimals(this.lastBlockPrice);
+    return this.lastBlockPrice && this.convertBaseDecimalsToPriceFeedDecimals(this.lastBlockPrice);
   }
 
   getPriceFeedDecimals() {
     return this.priceFeedDecimals;
   }
 
   async update() {
+    // Read token0 and token1 precision from Uniswap contract.
+    const [token0Address, token1Address] = await Promise.all([
+      this.uniswap.methods.token0().call(),
+      this.uniswap.methods.token1().call()
+    ]);
+    this.token0 = new this.web3.eth.Contract(this.erc20Abi, token0Address);
+    this.token1 = new this.web3.eth.Contract(this.erc20Abi, token1Address);
+    const [token0Precision, token1Precision] = await Promise.all([
+      this.token0.methods.decimals().call(),
+      this.token1.methods.decimals().call()
+    ]);
+    // The base currency of the Uniswap pool pair, which is `token1` unless
+    // `invertPrice == True` is the precision that the internal method `_getPriceFromSyncEvent()`
+    // returns prices in, and `convertBaseDecimalsToPriceFeedDecimals` will convert from this precision
+    // to the user's desired `priceFeedDecimals`.
+    this.baseCurrencyDecimals = Number(this.invertPrice ? token0Precision : token1Precision);
+    this.quoteCurrencyDecimals = Number(this.invertPrice ? token1Precision : token0Precision);
```

Comment:
Just want to check my understanding.

Non inverted:
Quote Currency == token0
Base Currency == token1
price = token1 / token0
price decimals == token1 decimals (base?)

Inverted:
Quote Currency == token1
Base currency == token0
price decimals == token0 decimals (base?)


Is my understanding correct?

Reply:
Correct but `priceDecimals` in the context of this PriceFeed is equal the precision that the user wants to return prices. So, the `_getPriceFromSyncEvent()` returns prices in the same precision as the base currency (token1 unless inverted), and then that precision is converted into `priceDecimals`.

For example, say the base decimals is 8, but the user wants prices in 10, then the return value of `_getPriceFromSyncEvent()` will get scaled up by 10**2

Reply:
right. I agree with your point here nick. I think the key is that the decimals that the feed returns at will be different to that of the pair and should always be scaled by the `priceFeedDecimals`, irrespective of the number of decimals in the pooled tokens.